### PR TITLE
[Giga-net] Activate XVal for newly mined blocks

### PR DESCRIPTION
**Pull request by ptschip**

### CODE REVIEW

**Comment by sickpig** [comment]
@ptschip
If I understand correctly this change will turn on Xval for block that we are trying to mine, i.e. we are going to leverage Xval for the `getblocktemplate` and `getminingcandidate`.

Am I right?

**Comment by ptschip** [comment]
yes

**Review by sickpig — APPROVED** [review]

File: src/allowed_args.cpp [hunk 1]
```diff
@@ -678,7 +678,9 @@ static void addBlockCreationOptions(AllowedArgs &allowedArgs)
         .addArg("blockprioritysize=<n>", requiredInt,
             strprintf(_("Set maximum size of high-priority/low-fee transactions in bytes (default: %d)"),
                     DEFAULT_BLOCK_PRIORITY_SIZE))
-        .addArg("blockversion=<n>", requiredInt, _("Generated block version number.  Value must be an integer"));
+        .addArg("blockversion=<n>", requiredInt, _("Generated block version number.  Value must be an integer"))
+        .addDebugArg(
```
Comment (gandrewstone) [inline src/allowed_args.cpp]: If this was a tweak we could change it dynamically which would be very useful.

Comment (ptschip) [reply src/allowed_args.cpp]: done

**Comment by sickpig** [comment]
@ptschip need rebase

**Comment by ptschip** [comment]
> @ptschip need rebase

rebased

**Review by sickpig — APPROVED** [review]

**Pull request merged**